frontend/components/SampleMutationRequest.tsx

```diff
@@ -0,0 +1,83 @@
+// Sample Mutation to be used for reference (for POST, PUT, DELETE)
+
+import React, { useState } from 'react';
+import axios from 'axios';
+import { Card, Text, Button } from 'tamagui';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useAuth } from '@clerk/clerk-react';
+
+export default function SampleMutationRequest() {
+  const [mutationResult, setMutationResult] = useState(null);
+  const [error, setError] = useState(null);
+  const { getToken } = useAuth();
+
+  // Make POST request with axios
+  const createPost = async (postData) => {
+    try {
+      const token = await getToken();
+      console.log(token);
+      const response = await axios.post(
+        `http://${process.env.EXPO_PUBLIC_IP_ADDR}/api/posts`,
+        postData,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        },
+      );
+      if (response.status === 201) {
+        return response.data;
+      } else {
+        throw new Error(
+          `Request failed with status ${response.status} and token ${token}`,
+        );
+      }
+    } catch (error) {
+      throw new Error(error.message);
+    }
+  };
+
+  const mutation = useMutation({
+    // mutationKey: ['UniqueNameSpecificToMutation'], // Optional: Descriptive key to identify this specific mutation
+    mutationFn: createPost, // Function that defines how to fetch data for this mutation
+  });
+
+  // on submit, send HTTP request
+  const handleSubmit = async () => {
+    if (mutation.isPending) return; // Prevent multiple submissions
+    try {
+      const data = await mutation.mutateAsync({
+        author: 'users/user_2cwMgsX7SwXnnnYJ2piefltKxLO',
```

Comment:
Should use the `token` variable from above instead of hardcoding.

Reply:
User token is different than session token for auth, do we have a specific user id that we might need to fetch?

Reply:
NVM, see changes